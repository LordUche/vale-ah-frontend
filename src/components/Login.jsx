import React from 'react';

/**
 * Renders a basic Login component
 * @returns {JSX} h2 element
 */
const Login = () => {
  return <h2>Kindly login below</h2>;
};

export default Login;
