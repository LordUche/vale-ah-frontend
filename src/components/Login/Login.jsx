import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import Auth from '../Auth/Auth';
import { handleMessages } from '../../utils/helpers';

/**
 * @param {object} props
 * @returns {JSX.Element} Login component
 */
const Login = props => {
  const { onClose, success, isLoading } = props;
  if (success) {
    handleMessages(['Login Successful'], 'success');
    onClose();
    return <Redirect to="/recipes/create" />;
  }

  return <Auth authType="login" onClose={onClose} isLoading={isLoading} />;
};

Login.defaultProps = {
  isLoading: false,
};

Login.propTypes = {
  onClose: PropTypes.func.isRequired,
  isLoading: PropTypes.bool,
  success: PropTypes.bool.isRequired,
};

/**
 *
 * @param {*} state
 * @return {undefined}
 */
const mapStateToProps = state => ({
  ...state.auth,
});

export default connect(mapStateToProps)(Login);
