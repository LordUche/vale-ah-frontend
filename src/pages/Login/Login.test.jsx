import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Login from './Login';

configure({ adapter: new Adapter() });
describe('<App />', () => {
  it('h1 contains correct text', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.find('h2').text()).toBe('Kindly login below');
  });
  it('matches the snapshot', () => {
    const tree = shallow(<Login />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
