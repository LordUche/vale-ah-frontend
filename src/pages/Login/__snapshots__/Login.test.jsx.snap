// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<h2>
  Kindly login below
</h2>
`;
